oap-server/analyzer/agent-analyzer/src/main/java/org/apache/skywalking/oap/server/analyzer/provider/trace/sampling/SamplingPolicySettingsReader.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.skywalking.oap.server.analyzer.provider.trace.sampling;
+
+import org.apache.skywalking.apm.util.StringUtil;
+import org.yaml.snakeyaml.Yaml;
+import org.yaml.snakeyaml.constructor.SafeConstructor;
+
+import java.io.InputStream;
+import java.io.Reader;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * SamplePolicySettingsReader parses the given `trace-sampling-policy-settings.yml` config file, to the target {@link SamplingPolicySettings}.
+ */
+public class SamplingPolicySettingsReader {
+    private Map yamlData;
+
+    public SamplingPolicySettingsReader(InputStream inputStream) {
+        Yaml yaml = new Yaml(new SafeConstructor());
+        yamlData = (Map) yaml.load(inputStream);
+    }
+
+    public SamplingPolicySettingsReader(Reader io) {
+        Yaml yaml = new Yaml(new SafeConstructor());
+        yamlData = (Map) yaml.load(io);
+    }
+
+    /**
+     * Read policy config file to {@link SamplingPolicySettings}
+     */
+    public SamplingPolicySettings readSettings() {
+        SamplingPolicySettings samplingPolicySettings = new SamplingPolicySettings();
+        if (Objects.nonNull(yamlData)) {
+            readDefaultSamplingPolicy(samplingPolicySettings);
+            readServicesSamplingPolicy(samplingPolicySettings);
+        }
+        return samplingPolicySettings;
+    }
+
+    private void readDefaultSamplingPolicy(SamplingPolicySettings samplingPolicySettings) {
+        Map<String, Object> objectMap = (Map<String, Object>) yamlData.get("default");
+        if (objectMap == null) {
+            return;
+        }
+        if (objectMap.get("rate") != null) {
+            samplingPolicySettings.getDefaultPolicy().setRate((Integer) objectMap.get("rate"));
+        }
+        if (objectMap.get("duration") != null) {
+            samplingPolicySettings.getDefaultPolicy().setDuration((Integer) objectMap.get("duration"));
+        }
```

Comment:
Here is one caveat: when the settings is configured via dynamic configuration, and the configured policy is 

```yaml
default:
  rate: 123
  duration: non-a-number
```

The default policy is **partially** updated, i.e., the `rate` is updated while the `duration` is not updated because it has an invalid value, and if I were a user, I'd expect the `rate` and `duration` to be updated both or none

Reply:
I had run the Uts for the case ,it is ok.  
![image](https://user-images.githubusercontent.com/9264192/131281397-759ad3bf-aeb9-47c9-b2fe-6d4486eaad0c.png)

The same as :
![image](https://user-images.githubusercontent.com/9264192/131281558-29728099-589a-4981-b31e-13ca515a0ff1.png)
![image](https://user-images.githubusercontent.com/9264192/131281592-0b6a0db8-51ab-47d5-b568-42ab75d9bc26.png)

Reply:
I think you should use `"abc"` in the YAML?

Reply:
@HendSame From the code reading, I feel Zhenxu is correct. If `duration` is invalid, then `.setDuration((Integer) objectMap.get("duration"))` would fail. In this case, once `rate` has been updated, then we would face partial-update case.

Could you recheck about this?

Reply:
> @HendSame From the code reading, I feel Zhenxu is correct. If `duration` is invalid, then `.setDuration((Integer) objectMap.get("duration"))` would fail. In this case, once `rate` has been updated, then we would face partial-update case.
> 
> Could you recheck about this?

It's indeed not partially updated, which is kind of tricky because if `org.apache.skywalking.oap.server.analyzer.provider.trace.TraceSamplingPolicyWatcher#parseFromYml` throws exception (`ClassCastException` from `String` `abc` to `Integer`), it returns `null` and thus `onUpdated(parseFromYml(config));` won't be executed, so the settings remain to the last update...

Reply:
Oh, I see. This is a tricky flow control. I thought it should be something inside method.
OK, I would not require this has to be changed, @kezhenxu94 your call.

Reply:
@wu-sheng @kezhenxu94 
I think it is ok.Is there anything else I haven't considered? I need your help. 
![image](https://user-images.githubusercontent.com/9264192/131295181-8c79861b-c705-44c2-aac8-9ed8540687bc.png)
![image](https://user-images.githubusercontent.com/9264192/131295254-187ae4ab-916c-4014-94f7-df6bbffe5fdc.png)